### PR TITLE
feat: Create SMTPConnectionResponseError

### PR DESCRIPTION
**Pull request by ikrivosheev**
@cole hello! Thank you for the library!

In my project I lost information from SMTP server. I need to know SMTP response code if I get `SMTPConnectionError`. I added `SMTPConnectResponseError` with `code` and `message` from SMTP server.

### CODE REVIEW

File: aiosmtplib/errors.py [hunk 1]
```diff
@@ -78,6 +79,16 @@ def __init__(self, code: int, message: str) -> None:
         self.args = (code, message)
 
 
+class SMTPConnectResponseError(SMTPResponseException, SMTPConnectError):
+    """
+    An error occurred while connecting to the SMTP server.
```
Comment (cole) [inline aiosmtplib/errors.py:84]: How about "The SMTP server returned an invalid response code after connecting." or similar so it's a bit more clear in the docs?

File: aiosmtplib/errors.py [hunk 2]
```diff
@@ -78,6 +79,16 @@ def __init__(self, code: int, message: str) -> None:
...
+    """
+
+    def __init__(self, code: int, message: str) -> None:
```
Comment (cole) [inline aiosmtplib/errors.py:87]: I think we don't need this method, if you just leave it out SMTPResponseException's init will be used.

**Review by cole — CHANGES_REQUESTED** [review]
Hi, thanks for your PR! Few minor things noted.

File: aiosmtplib/errors.py (hunk 1 above)
Comment (ikrivosheev) [reply aiosmtplib/errors.py]: Done!

File: aiosmtplib/errors.py (hunk 2 above)
Comment (ikrivosheev) [reply aiosmtplib/errors.py]: Done!

**Comment by ikrivosheev** [comment]
> Hi, thanks for your PR! Few minor things noted.

@cole thank you for the review, done.

**Review by cole — APPROVED** [review]

**Pull request merged**